src/main/java/com/shapesecurity/salvation/directives/Directive.java

```diff
@@ -19,6 +20,43 @@
 
     @Nonnull private Set<Value> values;
 
+    public static List<Class<? extends Directive>> getFetchDirectives() {
+        return fetchDirectives;
+    }
+
+    static List<Class<? extends Directive>> fetchDirectives = new ArrayList<>();
+
+    static void register(Class<? extends Directive> directiveClass) {
+        if (FetchDirective.class.isAssignableFrom(directiveClass) && directiveClass != DefaultSrcDirective.class) {
+            fetchDirectives.add(directiveClass);
+        }
+    }
+
+    static {
```

Comment:
We might want to make a NOTE here that new directives must be added to this list.